idseq_dag/util/s3.py

```diff
@@ -149,6 +159,26 @@ def fetch_from_s3(src,
                 return None
 
 
+def fetch_from_s3_whichever_exists(possible_sources,
+                                   dst,
+                                   auto_unzip=DEFAULT_AUTO_UNZIP,
+                                   auto_untar=DEFAULT_AUTO_UNTAR,
+                                   allow_s3mi=DEFAULT_ALLOW_S3MI):
+    possible_sources = list(possible_sources)
+    for i, src in enumerate(possible_sources):
+        try:
+            return fetch_from_s3(src, dst, auto_unzip=auto_unzip, auto_untar=auto_untar, allow_s3mi=allow_s3mi)
+        except:
```

Comment:
Note there is no backoff here because the reason for an exception is almost certainly that the file doesn't exist, rather than a system reason.